acceptance/localcluster/localcluster.go

```diff
@@ -0,0 +1,297 @@
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Peter Mattis (peter.mattis@gmail.com)
+
+package localcluster
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/samalba/dockerclient"
+)
+
+const (
+	builderImage   = "cockroachdb/builder"
+	dockerspyImage = "cockroachdb/docker-spy"
+	domain         = "local"
+)
+
+var cockroachImage = flag.String("i", builderImage, "the docker image to run")
+var cockroachBinary = flag.String("b", defaultBinary(), "the binary to run (if image == "+builderImage+")")
+var cockroachEntry = flag.String("e", "", "the entry point for the image")
+
+func prettyJSON(v interface{}) string {
+	pretty, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	return string(pretty)
+}
+
+func defaultBinary() string {
+	gopath := filepath.SplitList(os.Getenv("GOPATH"))
+	if len(gopath) == 0 {
+		return ""
+	}
+	return gopath[0] + "/bin/linux_amd64/cockroach"
+}
+
+func exists(path string) bool {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+// Cluster manages a local cockroach cluster running on docker. The cluster is
+// composed of a "dns" container which automatically registers dns entries for
+// the cockroach nodes, a "volumes" container which manages the persistent
+// volumes used for certs and node data and N cockroach nodes.
+type Cluster struct {
+	client  dockerclient.Client
+	dns     *Container
+	vols    *Container
+	Nodes   []*Container
+	stopper chan struct{}
+}
+
+// Create creates a new local cockroach cluster. The stopper is used to
+// gracefully shutdown the channel (e.g. when a signal arrives). The cluster
+// must be started before being used.
+func Create(numNodes int, stopper chan struct{}) *Cluster {
+	if *cockroachImage == builderImage && !exists(*cockroachBinary) {
+		log.Fatalf("\"%s\": does not exist", *cockroachBinary)
+	}
+
+	return &Cluster{
+		client:  newDockerClient(),
+		Nodes:   make([]*Container, numNodes),
+		stopper: stopper,
+	}
+}
+
+func (l *Cluster) stopOnPanic() {
+	if r := recover(); r != nil {
+		l.Stop()
+		if r != l {
+			panic(r)
+		}
+	}
+}
+
+func (l *Cluster) panicOnStop() {
+	if l.stopper == nil {
+		panic(l)
+	}
+
+	select {
+	case <-l.stopper:
+		l.stopper = nil
+		panic(l)
+	default:
+	}
+}
+
+func (l *Cluster) runDockerSpy() {
+	l.panicOnStop()
+
+	create := func() (*Container, error) {
+		return createContainer(l.client, dockerclient.ContainerConfig{
+			Image: dockerspyImage,
+			Cmd:   []string{"--dns-domain=" + domain},
+		})
+	}
+	c, err := create()
+	if err == dockerclient.ErrNotFound {
+		log.Printf("pulling %s", dockerspyImage)
+		err = l.client.PullImage(dockerspyImage, nil)
+		if err == nil {
+			c, err = create()
+		}
+	}
+	if err != nil {
+		panic(err)
+	}
+	maybePanic(c.Start([]string{"/var/run/docker.sock:/var/run/docker.sock"}, nil, nil))
```

Comment:
I don't know. You should test it on linux and see if it works for you. This mapping was what docker-spy told me to do: https://github.com/iverberk/docker-spy